### PR TITLE
Improving filtering for hephaestus

### PR DESCRIPTION
**Pull request by wildunne**
## What:

For hephaestus we need to apply some filtering to hide things like passwords and PANs from raygun. Currently the filtering done here doesn't actually clean for example the request data, so this change is designed to fix that.

@AntonTutuka @ewanmttk please review

### CODE REVIEW

File: src/nz/co/ventego-creative/raygun4cfml/RaygunClient.cfc [hunk 1]
```diff
@@ -51,6 +51,7 @@ limitations under the License.
 			var messageContent = "";
 			var jSONData = "";
 			var postResult = "";
+            var doClean = false;
```
Comment (ewanmttk) [inline src/nz/co/ventego-creative/raygun4cfml/RaygunClient.cfc]: Check indents

File: src/nz/co/ventego-creative/raygun4cfml/RaygunClient.cfc [hunk 2]
```diff
@@ -66,9 +67,10 @@ limitations under the License.
 				throw("API integration not valid, cannot send message to Raygun");
 			}
 
-            if (isObject(variables.contentFilter))
+            if (isObject(variables.contentFilter) && isArray(arguments.contentFilter) && arraylen(arguments.contentFilter) > 0)
```
Comment (ewanmttk) [inline src/nz/co/ventego-creative/raygun4cfml/RaygunClient.cfc]: ContentFilter is defined in variables, not arguments.

File: src/nz/co/ventego-creative/raygun4cfml/RaygunClient.cfc [hunk 3]
```diff
@@ -102,6 +104,10 @@ limitations under the License.
             // KK: This will only work if the users has setup none or the default prefix for JSON data
             jSONData = ReplaceNoCase(trim(jSONData), "//{", "{");
             jSONData = ReplaceNoCase(trim(jSONData), "//[", "[");
+
+            if(doClean) {
+                jSONData = applyFilterJson(arguments.contentFilter, jSONData);
```
Comment (ewanmttk) [inline src/nz/co/ventego-creative/raygun4cfml/RaygunClient.cfc]: And same here - should be variables.contentFilter

File: src/nz/co/ventego-creative/raygun4cfml/RaygunClient.cfc (hunk 1 above)
Comment (AntonTutuka) [inline src/nz/co/ventego-creative/raygun4cfml/RaygunClient.cfc]: is indent ok here?

File: src/nz/co/ventego-creative/raygun4cfml/RaygunClient.cfc (hunk 2 above)
Comment (AntonTutuka) [inline src/nz/co/ventego-creative/raygun4cfml/RaygunClient.cfc]: It's better to use the same variable `variables.contentFilter`, not mixing it with arguments.contentFilter

**Comment by wildunne** [comment]
Ah, good spot, didn't show up in my editor

**Pull request merged**